core-java-security/src/main/java/com/baeldung/ssl/EnableTLSv12.java

```diff
@@ -0,0 +1,134 @@
+package com.baeldung.ssl;
+
+import java.io.BufferedReader;
+import java.io.BufferedWriter;
+import java.io.IOException;
+import java.io.InputStreamReader;
+import java.io.OutputStreamWriter;
+import java.io.PrintWriter;
+import java.net.URL;
+import java.security.SecureRandom;
+
+import javax.net.ssl.HttpsURLConnection;
+import javax.net.ssl.SSLContext;
+import javax.net.ssl.SSLParameters;
+import javax.net.ssl.SSLSocket;
+import javax.net.ssl.SSLSocketFactory;
+
+public class EnableTLSv12 {
+
+    public String url = "";
+    public Integer port = null;
+
+    public EnableTLSv12() {
+    }
+
+    public static void main(String[] args) {
+        try {
+            EnableTLSv12 enableTLSv12 = new EnableTLSv12();
+            if (args.length != 2) {
+                System.out.println("Provide the server url and the secure port:");
+                System.exit(-1);
+            }
+            enableTLSv12.getHost(args);
+            enableTLSv12.getPort(args);
+            enableTLSv12.enableTLSv12UsingHttpConnection();
+            enableTLSv12.enableTLSv12UsingProtocol();
+            enableTLSv12.enableTLSv12UsingSSLContext();
+            enableTLSv12.enableTLSv12UsingSSLParameters();
+        } catch (Exception e) {
+            e.printStackTrace();
+        }
+
+    }
+
+    private void getPort(String[] args) {
+        url = args[0];
+    }
+
+    private void getHost(String[] args) {
+        String portNumber = args[1];
+        port = Integer.parseInt(portNumber);
+    }
+
+    private void handleCommunication(SSLSocket socket, String usedTLSProcess) throws IOException {
+        System.out.print("Enabled TLS v1.2 on " + usedTLSProcess);
+        PrintWriter out = new PrintWriter(new BufferedWriter(new OutputStreamWriter(socket.getOutputStream())));
+        out.println("GET / HTTP/1.0");
+        out.println();
+        out.flush();
+        if (out.checkError())
+            System.out.println("SSLSocketClient:  java.io.PrintWriter error");
+
+        BufferedReader in = new BufferedReader(new InputStreamReader(socket.getInputStream()));
+
+        String inputLine;
+        while ((inputLine = in.readLine()) != null)
+            System.out.println(inputLine);
+
+        in.close();
```

Comment:
These won't get closed in the case of an exception, right? We want to try and show good engineering practice, even though closing streams isn't so important in an example like this.

`try-with-resources` is available in Java 7, so I think these could be cleaned up nicely.